client/pkg/transport/listener.go

```diff
@@ -66,7 +70,7 @@ func newListener(addr, scheme string, opts ...ListenerOption) (net.Listener, err
 		lnOpts.ListenConfig = config
 		// check for timeout
 		fallthrough
-	case lnOpts.IsTimeout(), lnOpts.IsSocketOpts():
+	case lnOpts.IsTimeout() && lnOpts.IsSocketOpts():
```

Comment:
I believe the existing implementation is correct, the logic should be `or` instead of `and`. 

Note that the previous case has a `fallthrough` at the end.